dev-qt/qt-creator/qt-creator-9999.ebuild

```diff
@@ -112,7 +112,17 @@ unset x
 PATCHES=( "${FILESDIR}"/${PN}-4.12.0-dylib-fix.patch )
 
 pkg_setup() {
-	use clang && llvm_pkg_setup
+	if use clang; then
+		llvm_pkg_setup
+
+		# Find and export the clang slot that satisfies the dependency
+		local clang_slot=${LLVM_MAX_SLOT}
+		while [[ ! -x "$(get_llvm_prefix ${clang_slot})/bin/clang" ]]; do
+			# No need for lower-bounds if clang dep is satisfied
+			clang_slot=$((clang_slot - 1))
+		done
+		export CLANG_PREFIX="$(get_llvm_prefix ${clang_slot})"
```

Comment:
I'm not sure I understand this patch, are you trying to reimplement `llvm_check_deps()`?

Reply:
Yes, it appears I was.  I should pay more attention to the bash comments.  Fixed.